### PR TITLE
Separate es2 from redis

### PR DESCRIPTION
**Pull request by kishorebanala**
- Created es2-persistence standalone module.
- Removed Elasticsearch dependencies from redis-persistence module.
- Provided config options to switch between ES2 and ES5 modules.
- Used shadowing on es2-persistence module to resolve dependency conflicts.
- Other cleanup and refactoring.

Breaking changes:
Current implementation provides ES bindings from redis-persistence module. Now that es2-persistence is a separate module, it (or es5-persistence) has to be imported to Server module. Also, es2 dependencies are shaded and relocated to `conductor.` extensions;

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/dao/IndexDAO.java [hunk 1]
```diff
@@ -164,4 +164,12 @@
 	 */
 	void addMessage(String queue, Message msg);
 
+	/**
+	 *
+	 * @param indexName Name of the index to search
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/dao/IndexDAO.java]: javadoc

File: es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java [hunk 2]
```diff
@@ -472,4 +474,26 @@ public String get(String workflowInstanceId, String fieldToGet) {
 		long count = response.getHits().getTotalHits();
 		return new SearchResult<String>(count, result);
 	}
+
+	@Override
+	public List<String> searchArchivableWorkflows(String indexName, long archiveTtlDays) {
+		QueryBuilder q = QueryBuilders.boolQuery()
+				.must(QueryBuilders.rangeQuery("endTime").lt(LocalDate.now().minusDays(archiveTtlDays)))
+				.must(QueryBuilders.termQuery("status", "COMPLETED"))
+				//.should(QueryBuilders.termQuery("status", "FAILED"))
```
Comment (cyzhao) [inline es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java]: uncomment this

File: redis-persistence/src/main/java/com/netflix/conductor/dao/RedisWorkflowModule.java [hunk 3]
```diff
@@ -32,14 +31,18 @@
 
 /**
  * @author Viren
+ * <p>
```
Comment (cyzhao) [inline redis-persistence/src/main/java/com/netflix/conductor/dao/RedisWorkflowModule.java]: remove this comment

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java [hunk 4]
```diff
@@ -468,4 +477,27 @@ public String get(String workflowInstanceId, String fieldToGet) {
...
 		return new SearchResult<>(count, result);
 	}
+
+
+	@Override
+	public List<String> searchArchivableWorkflows(String indexName, long archiveTtlDays) {
+		QueryBuilder q = QueryBuilders.boolQuery()
+				.must(QueryBuilders.rangeQuery("endTime").lt(LocalDate.now().minusDays(archiveTtlDays)))
+				.must(QueryBuilders.termQuery("status", "COMPLETED"))
+				//.should(QueryBuilders.termQuery("status", "FAILED"))
```
[the remark this answers is not in the document]
Comment (apanicker-nflx) [reply es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: also `must` -> `should` in the line above.

File: server/src/main/java/com/netflix/conductor/server/ConductorServer.java [hunk 5]
```diff
@@ -147,7 +148,13 @@ private void init(String dynoClusterName, List<Host> dynoHosts) {
 		case memory:
 			jedis = new JedisMock();
 			try {
-				EmbeddedElasticSearch.start();
+				if (conductorConfig.getProperty("workflow.elasticsearch.version", "2").startsWith("5")){
```
Comment (cyzhao) [inline server/src/main/java/com/netflix/conductor/server/ConductorServer.java]: change startsWith to equals

File: server/src/main/java/com/netflix/conductor/server/ServerModule.java [hunk 6]
```diff
@@ -80,19 +70,24 @@ public ServerModule(JedisCommands jedis, HostSupplier hostSupplier, ConductorCon
...
 		if (db == ConductorServer.DB.mysql) {
 			install(new MySQLWorkflowModule());
 		} else {
-		    install(new RedisESWorkflowModule(conductorConfig, dynoConn, hostSupplier));
+			install(new RedisWorkflowModule(conductorConfig, dynoConn, hostSupplier));
 		}
 
-		install(new ElasticsearchModule());
-		bind(IndexDAO.class).to(ElasticSearchDAO.class);
+		if (conductorConfig.getProperty("workflow.elasticsearch.version", "2").startsWith("5")){
```
Comment (cyzhao) [inline server/src/main/java/com/netflix/conductor/server/ServerModule.java]: startsWith to equals

**Pull request merged**